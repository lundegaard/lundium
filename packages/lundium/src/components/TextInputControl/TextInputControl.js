import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';

import FormControl from '../FormControl';

const TextInputControl = forwardRef(({ children, className, type = 'text', ...rest }, ref) => (
	<FormControl className={cx(className)} component="input" type={type} ref={ref} {...rest}>
		{children}
	</FormControl>
));

TextInputControl.displayName = 'forwardRef(TextInputControl)';

TextInputControl.propTypes = {
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	/** Class name for root. */
	className: PropTypes.string,
	/** HTML type attribute of the input. */
	type: PropTypes.string,
};

export default TextInputControl;
