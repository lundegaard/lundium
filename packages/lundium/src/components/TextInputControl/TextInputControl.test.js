import React from 'react';
import { shallow } from 'enzyme';

import TextInputControl from './TextInputControl';

describe('<TextInputControl />', () => {
	const requiredProps = {};

	it('shallow', () => {
		const wrapper = shallow(<TextInputControl {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
