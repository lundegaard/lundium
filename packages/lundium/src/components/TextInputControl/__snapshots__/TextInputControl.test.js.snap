// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInputControl /> shallow 1`] = `
<FormControl
  className=""
  component="input"
  onBlur={[Function]}
  onFocus={[Function]}
  type="text"
/>
`;
